src/libraries/libraries-packages.proj

```diff
@@ -90,8 +84,20 @@
     <Message Condition="'@(_PackageReports)' != '' and Exists('$(ArtifactsDir)packages')" Importance="High" Text="##vso[task.setvariable variable=_librariesBuildProducedPackages]true" />
   </Target>
 
+  <Target Name="EnableBuildPackage">
+    <MSBuild Projects="@(ProjectReference)"
+             Targets="GetServicePackage"
+             BuildInParallel="$(BuildInParallel)">
+      <Output TaskParameter="TargetOutputs" ItemName="ProjectReference" />
```

Comment:
Dynamic dependencies aren't supported by msbuild static graph hence this will introduce yet another blocker to adopt static graph build (static graph restore is already enabled).

I think we should instead use existing shipping packaging infrastructure from the SDK which allows a project to set `IsPackable` to control if a package should be generated.

Reply:
Yeah using `IsPackable` would be more intuitive as we could just set `IsPackable=false` in `Directory.Build.props` and then packages that we need to ship we would set them as `IsPackable=true`.

Reply:
And as I just saw that this PR targets release/5.0, we should do the same in master then.

Reply:
I gave this more thought and I believe I was wrong when I said that `IsPackable` should be used in servicing cases. `IsPackable` determines if a project is packable per-se, which controls if the Pack target is invokable on projects that import NuGet target files.

During the transition from pkgproj to csproj files, csproj files will be annotated with `<IsPackable>true</IsPackable>` to express that they are packable via the NuGet pack task instead of via the Microsoft.Dotnet.Built.Task.Packaging` package. Overloading the semantic of `IsPackable` with `IsServiceable` will cause problems onwards, hence I believe the addition `ServicePackabe` property makes sense.

Reply:
I implemented your proposal in master via https://github.com/dotnet/runtime/pull/46447. It allows filtering on a project level instead of at traversal time. A nice side-effect of this work is that incremental servicing via solution files now works as well (instead of files that use Microsoft.Build.Traversal SDK in which you can apply conditions per P2P).